deploy/setup.sh

```diff
@@ -0,0 +1,280 @@
+#!/bin/bash
+:<<BLOCK
+Copyright (c) Microsoft Open Technologies (Shanghai) Co. Ltd.  All rights reserved.
+
+The MIT License (MIT)
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+BLOCK
+pre_setup_docker() {
+    #for ubuntu 14
+    if $(lsb_release -d | grep -q "14"); then
+        echo "deb https://apt.dockerproject.org/repo ubuntu-trusty main" | tee /etc/apt/sources.list.d/docker.list
+    fi
+
+    # for ubuntu 15
+    if $(lsb_release -d | grep -q "15"); then
+        echo "deb https://apt.dockerproject.org/repo ubuntu-wily main" | tee /etc/apt/sources.list.d/docker.list
+    fi
+
+    # for linux mint
+    if $(lsb_release -d | grep -q "Mint"); then
+        echo "deb https://apt.dockerproject.org/repo ubuntu-trusty main" | tee /etc/apt/sources.list.d/docker.list
+    fi
+
+    apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D
+}
+
+get_dependency_software() {
+    echo "updating apt-get......"
+    result=$(sudo apt-get update)
+    if grep -q "Could not resolve" <<< $result; then
+        echo "Could not update apt-get, please solve it"
+        exit
+    fi
+    echo "installing git python-setuptools python-dev python-pip"
+    result=$(sudo apt-get install -y git python-setuptools python-dev python-pip)
+    if grep -q "Unable to lacate" <<< $result; then
+        echo "Could not install dependancy software, pls install docker manually"
+        exit
+    fi
+    echo "installing autoconf libtool tomcat7 "
+    result=$(sudo apt-get install -y autoconf automake libtool tomcat7 )
+    if grep -q "Unable to lacate" <<< $result; then
+        echo "Could not install dependancy software, pls install docker manually"
+        exit
+    fi
+}
+
+set_envirement() {
+    echo "git clone source code from Github, and solve python dependency"
+    cd /home &&  rm -rf opentech
+    mkdir opentech && cd opentech
+    rm -r open-hackathon
```

Comment:
Do we need `sudo` to run this script? I guess so since there are many `sudo` in the scripts. In this case, the directory `open-hackathon` is owned by `root` which is a bad practice. Please change the owner/group to `opentech` recursively.